### PR TITLE
Feature release guide

### PR DESCRIPTION
**Pull request by jprestop**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>

Reviewed the changes here:
https://metplus.readthedocs.io/en/feature_release_guide/Release_Guide/met_development.html
Other pages are affected as well, but just listed this one for quick reference and formatting.  

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Review the files changed for proper wording and typos.  

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[N/A]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **[09/10/21]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [N/A] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Release_Guide/release_steps/common/update_dtc_website.rst [hunk 1]
```diff
@@ -47,4 +46,15 @@ Update DTC Website
 
   * Click on "Save".
 
+  * Update the existing releases, as needed.
+    
+    * For a development release, change any previous "Development"
```
Comment (JohnHalleyGotway) [inline docs/Release_Guide/release_steps/common/update_dtc_website.rst]:
```suggestion
    * For a development release, change any previous *Development*
```

File: docs/Release_Guide/release_steps/common/update_dtc_website.rst [hunk 2]
```diff
@@ -47,4 +46,15 @@ Update DTC Website
...
+      versions to "Other".
```
Comment (JohnHalleyGotway) [inline docs/Release_Guide/release_steps/common/update_dtc_website.rst]:
```suggestion
      versions to *Other*.
```

File: docs/Release_Guide/release_steps/common/update_dtc_website.rst [hunk 3]
```diff
@@ -47,4 +46,15 @@ Update DTC Website
...
+      
+    * For a bugfix or official release, change any previous
+      "Recommended" versions to "Other".
```
Comment (JohnHalleyGotway) [inline docs/Release_Guide/release_steps/common/update_dtc_website.rst]:
```suggestion
      *Recommended* versions to *Other*.
```

File: docs/Release_Guide/release_steps/common/update_dtc_website.rst (hunk 1 above)
Comment (JohnHalleyGotway) [reply docs/Release_Guide/release_steps/common/update_dtc_website.rst]: Updating formatting to follow italics conventions used above.

**Review by JohnHalleyGotway — CHANGES_REQUESTED** [review]
Proposed a few suggested edits to formatting.

It don't see the benefit of keeping "create_release_on_github.rst" and "create_release_extra.rst" as separate files. Wouldn't it be simpler to just add the 2 extra steps from "extra" onto the end of "create_release_on_github.rst"? And then remove the "extra" file and all references to it?

Also, I'm confused by "Release_Guide/release_steps/common/update_dtc_website.rst". Do you recall why we have it setup this way and how it relates to "Release_Guide/release_steps/update_dtc_website.rst"?

**Comment by jprestop** [comment]
@JohnHalleyGotway 
> I don't see the benefit of keeping "create_release_on_github.rst" and "create_release_extra.rst" as separate files. Wouldn't it be simpler to just add the 2 extra steps from "extra" onto the end of "create_release_on_github.rst"? And then remove the "extra" file and all references to it?

It's because METplus has this "extra" step (see below) that needs to be inserted before the other two.  I noticed that all of the other component's "extra" file were the same, so I brought them out of their subdirectories and put them into the main "release_steps" directory.

> Add link to METplus_Data and add a note to instruct users to use sub-directory corresponding to this release, i.e. v4.0.0)

> Also, I'm confused by "Release_Guide/release_steps/common/update_dtc_website.rst". Do you recall why we have it setup this way and how it relates to "Release_Guide/release_steps/update_dtc_website.rst"?

I don't recall the exactly why it was set it up that way.  I think it is because much of the text was the same for all components, but there were some variables (e.g. |projectRepo|, |addTarfileStep|, |otherWebsiteUpdates|) that needed to be replaced for each repo.  But I can't remember why it is in "common" as opposed to being in the general "release_steps" area with the other common files.   Perhaps @georgemccabe add additional information.

**Comment by georgemccabe** [comment]
release_steps/common/update_dtc_website.rst contains references to variables that contain text that is specific to a given component. Each component's version includes this file and overrides some variables to include the component-specific content. It was designed to reduce duplicate content while enabling slight customization for components.

It looks like release_steps/update_dtc_website.rst is not being used. It was added to be used by components that don't need any information changed from the "default" but it looks like each component has their own file. It looks like MET and METplus are the only components that have any content that differs from the default, so the following files could be removed in favor of using release_steps/update_dtc_website.rst:

release_steps/metcalcpy/update_dtc_website.rst
release_steps/metviewer/update_dtc_website.rst
release_steps/metplotpy/update_dtc_website.rst

I recall none of us really loved the name "common" for the directory but we decided to keep it as is and change it later. I was hoping to set the "default" variables in the file with the text content and override the values in other files as needed, but that produced an error saying that the variable was defined twice.

**Comment by jprestop** [comment]
@georgemccabe 

> It looks like MET and METplus are the only components that have any content that differs from the default, so the following files could be removed in favor of using release_steps/update_dtc_website.rst:
> 
> release_steps/metcalcpy/update_dtc_website.rst
> release_steps/metviewer/update_dtc_website.rst
> release_steps/metplotpy/update_dtc_website.rst

Nice catch!  Thanks!  I just removed the specific ones and pointed to the version in release_steps.  Thanks!

**Review by JohnHalleyGotway — APPROVED** [review]
Thanks for the discussion and changes @jprestop and @georgemccabe.

I approve of these changes.

Please merge when you see fit.

**Pull request merged**